code/frameworks/nextjs-server/src/verifyPort.ts

```diff
@@ -0,0 +1,49 @@
+import { join } from 'path';
+import { ensureDir, writeFile } from 'fs-extra';
+
+const writePidFile = async (pid: number, { appDir }: { appDir: boolean }) => {
+  const routeDir = appDir ? join('app', '(sb)') : 'pages';
+  const storybookDir = join(process.cwd(), routeDir, 'storybookPreview');
+  const pidFile = join(storybookDir, 'pid.tsx');
```

Comment:
I don't think this file name is right for the app dir?

Also this will overwrite the file every time we start, that's a bit icky. Can we do it with a middleware somehow?